resources/resource-extensbility/2024-06-resource-extensibility-feature-spec.md

```diff
@@ -0,0 +1,432 @@
+
+### Current state in Radius
+
+[Radius Extenders](https://docs.radapp.io/guides/author-apps/custom/overview/#extenders) enables users to author custom resource types but they are weakly types. They allow you to pass in any property or secret and for cases where the developer/operator need to extend and reference a simple resource type definition. Extenders are for untyped/ weakly typed resources and do not provide a lot of customization for users wanting to have guardrails in their platforms with strongly typed resource definitions following a strict set of rules and restrictions. 
+
+## Existing customer problem
+<!-- <Write this in first person. You basically want to summarize what “I” as a customer am trying to accomplish, why the current experience is a problem and the impact it has on me, my team, my work and or biz, etc…. i.e. “When I try to do x aspect of cloud native app development, I have the following challenges / issues….<details>. Those issues result in <negative impact those challenges / issues have on your work and or business.> -->
+
+### Customer Jobs to be done
+
+As a platform engineer trying to build a custom platform for application development, I have the following jobs to be done and requirements to meet:
+
+#### Main job : Build a custom platform for application development
+
+|  Related jobs | Functional aspects | Emotional aspects | 
+| ------------- | ------------------ | ----------------- | 
+| Define the set of key technologies for the enterprise applications | Set of technologies that organizations intend to use for their applications based on the nature of technology, open-source support, documentation support, existing vendor relations, expertise in the team | Easy to use, troubleshoot, sense of service ownership |
+| Specify the contract for the key services in the application platform | Author a schema for the custom resources easily with auto guidance tools | Simple tools that enables the schema authoring, uniformity and consistency |
+| Iterating and testing the application platform | Providing a productive inner loop for development and testing  | Simple and easy way to iterate on changes, test and deploy them |
+| Ship the custom platform | Registering the API's to the platform, providing documentation, onboarding and support | intuitive, seamless, collaborative |
+| Maintain the custom platform | Versioning and maintaining the API's | Ease of support, sense of ownership (decision making) | 
+| Support for compliance and security | Compatibility and security policies | Trust the platform |
+| Open-source custom resources | Repository and pipeline to validate, testing the infrastructure and publishing  | Desire to contribute, easy to use, low overhead /dependency to publish |
+
+As an IT operator and developer trying to build and maintain the application, I have the following jobs to be done and requirements to meet:
+
+#### Main job : Use the custom platform to build and maintain the application
```

Comment:
It looks like this includes the recipes as part of "use the platform". I'm still not sure about that... The recipes feel a little more like they are part of "build the platform". Not a super strong opinion.

Reply:
Moved it to as a separate job to avoid confusion.